spec/services/reports/monthly_bulk_report_service_spec.rb

```diff
@@ -5,14 +5,45 @@
 module Reports
   RSpec.describe MonthlyBulkReportService do
     describe ".run" do
-      it "generates a CSV file containing exemptions for the given month" do
-        monthly_bulk_serializer = double(:monthly_bulk_serializer)
+      let(:first_day_of_the_month) { Date.new(2019, 6, 1) }
+      context "when the request succeed" do
+        it "generates a CSV file containing exemptions for the given months and upload it to AWS" do
+          stub_successful_request
+          create_list(:registration_exemption, 2, :with_registration, registered_on: first_day_of_the_month)
 
-        expect(MonthlyBulkSerializer).to receive(:new).and_return(monthly_bulk_serializer)
-        expect(monthly_bulk_serializer).to receive(:to_csv)
+          expect_no_error
 
-        MonthlyBulkReportService.run(Date.today)
+          MonthlyBulkReportService.run(first_day_of_the_month)
+        end
       end
+
+      context "when the request succeed" do
```

Comment:
Think you have a copy+paste fail here. Guessing that should be `context "when the request fails"`

Reply:
Ops!! 🙈